sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,659 @@
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const createdMSFAANumber = await db.msfaaNumber.findOne({
+      select: {
+        id: true,
+      },
+      where: {
+        student: { id: student.id },
+      },
+    });
+    // Fetch MSFAA Number for the disbursement schedule assigned to the student in SIMS.
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
+      select: {
+        id: true,
+        msfaaNumber: { id: true },
+      },
+      relations: {
+        msfaaNumber: true,
+        studentAssessment: true,
+      },
+      where: {
+        studentAssessment: { id: application.currentAssessment.id },
+      },
+    });
+    // Assert MSFAA Number.
+    expect(disbursementSchedule.msfaaNumber.id).toBe(createdMSFAANumber.id);
+  });
+
+  it("Should reuse MSFAA Number for part time application when MSFAA is found for previously signed disbursement for part applications in SIMS.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    // Create and save fake MSFAA Number.
+    const savedMSFAANumber = await db.msfaaNumber.save(
+      createFakeMSFAANumber(
+        {
+          student: student,
+        },
+        {
+          msfaaState: MSFAAStates.Signed,
+          msfaaInitialValues: {
+            offeringIntensity: OfferingIntensity.partTime,
+            dateSigned: getISODateOnlyString(
+              addDays(-MAX_MSFAA_VALID_DAYS + 1),
+            ),
+          },
+        },
+      ),
+    );
+    // Create previous fake application with disbursements.
+    await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student, msfaaNumber: savedMSFAANumber },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+        firstDisbursementInitialValues: {
+          disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+        },
+        createSecondDisbursement: true,
+      },
+    );
+
+    // Create student and save fake application with disbursements.
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+        createSecondDisbursement: true,
+      },
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the disbursement schedule assigned to the student in SIMS.
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
```

Comment:
Since it was created with `createSecondDisbursement: true` both can be checked in the assert.